torchdata/stateful_dataloader/stateful_dataloader.py

```diff
@@ -940,17 +962,41 @@ def _reset(self, loader, first_iter=False, prime_prefetch=True):
         self._workers_status = [True for i in range(self._num_workers)]
         # Reset the worker queue cycle so it resumes next epoch at worker 0
         self._worker_queue_idx_cycle = itertools.cycle(range(self._num_workers))
-        # We resume the prefetching in case it was enabled
-        if not first_iter:
+        worker_states = {}
+        if first_iter:
```

Comment:
All logic for getting "initial" states from workers is here, for first iter and for resumes

Reply:
To make sure, I understand the outcome of the changes - for the case when load_state_dict is called before iter(dl), ie:

```
dl = StatefulDataLoader(...)
dl.load_state_dict(state_from_somewhere)
it = iter(dl)
print(next(it))
```

In this case, does AckStartup get sent out explicitly to get the initial worker state?

Reply:
Yes, AckStartup is always sent (My preference is to have fewer logic branches to keep the code understandable as much as possible

Reply:
So AckStartup is sent only once at the beginning of the first epoch. When iter(dl) is called subsequently, AckStartup is not sent, is that correct?

Reply:
It's sent on _reset as well